cmd/scollector/collectors/elasticsearch.go

```diff
@@ -35,189 +41,228 @@ func AddElasticIndexFilter(s string) error {
 	if err != nil {
 		return err
 	}
-	esIndexFilters = append(esIndexFilters, re)
+	elasticIndexFilters = append(elasticIndexFilters, re)
 	return nil
 }
 
+type structProcessor struct {
+	elasticVersion string
```

Comment:
is this too specific to elastic to generalize? It seems like struct tags to metrics is a really cool feature if we could make it generally available.